File: truevfs-kernel/truevfs-kernel-impl/src/main/scala/net/truevfs/kernel/impl/NeedsLockRetryException.scala
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package net.truevfs.kernel.impl

import de.schlichtherle.truecommons.shed._
import javax.annotation.concurrent._

/** Indicates that all file system locks need to get released before the
  * operation can get retried.
  *
  * @see    SyncController
  * @author Christian Schlichtherle
  */
@Immutable
private final class NeedsLockRetryException private ()
extends ControlFlowException

private object NeedsLockRetryException {
  import ControlFlowException._

  private[this] val instance = new NeedsLockRetryException

  def apply() = if (isTraceable) new NeedsLockRetryException else instance
}

File: src/main/java/global/namespace/truevfs/kernel/impl/NeedsLockRetryException.java
/*
 * Copyright © 2005 - 2021 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package global.namespace.truevfs.kernel.impl;

import global.namespace.truevfs.commons.shed.ControlFlowException;

/**
 * Indicates that all file system locks need to get released before the operation can get retried.
 *
 * @author Christian Schlichtherle
 * @see LockController
 */
final class NeedsLockRetryException extends ControlFlowException {

    private static final long serialVersionUID = 0;
    private static final NeedsLockRetryException instance = new NeedsLockRetryException();

    private NeedsLockRetryException() {
        super(false);
    }

    static NeedsLockRetryException apply() {
        return isTraceable() ? new NeedsLockRetryException() : instance;
    }
}
